File: 0776-n-ary-tree-postorder-traversal/0776-n-ary-tree-postorder-traversal.js
/**
 * // Definition for a _Node.
 * function _Node(val,children) {
 *    this.val = val;
 *    this.children = children;
 * };
 */

/**
 * @param {_Node|null} root
 * @return {number[]}
 */
var postorder = function(root) {
    const ans = [];
    if(root === null) return ans;

    const stack = [];
    stack.push(root);
    while(stack.length > 0) {
        const node = stack.pop();
        for(let child of node.children) {
            stack.push(child);
        }

        ans.push(node.val);
    }

    ans.reverse();
    return ans;
};

File: 0776-n-ary-tree-postorder-traversal/0776-n-ary-tree-postorder-traversal.java
/*
// Definition for a Node.
class Node {
    public int val;
    public List<Node> children;

    public Node() {}

    public Node(int _val) {
        val = _val;
    }

    public Node(int _val, List<Node> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
    public List<Integer> postorder(Node root) {
        List<Integer> ans = new ArrayList<>();
        if(root == null) return ans;

        Stack<Node> st = new Stack<>();
        st.push(root);

        while(!st.isEmpty()) {
            Node t = st.pop();
            for(Node child : t.children) st.push(child);
            ans.add(t.val);
        }

        Collections.reverse(ans);
        return ans;
    }
}